homeassistant/components/switch/mochad.py

```diff
@@ -50,24 +50,46 @@ def __init__(self, hass, ctrl, dev):
         self._comm_type = dev.get(mochad.CONF_COMM_TYPE, 'pl')
         self.device = device.Device(ctrl, self._address,
                                     comm_type=self._comm_type)
-        self._state = self._get_device_status()
+        """Init with false to avoid locking HA for long on CM19A"""
+        self._state = False
+        self.lock = threading.Lock()
```

Comment:
Should this be a lock shared among all devices?

Reply:
good point. I will need to move the lock creation up to the setup_platform(), then pass by reference to all devices.

Reply:
Ok. fixed in commit 3. e88301b